diff --git ./servers/rest_server/rest_server.info ./servers/rest_server/rest_server.info
index c05df4b..e2bee54 100644
--- ./servers/rest_server/rest_server.info
+++ ./servers/rest_server/rest_server.info
@@ -10,8 +10,8 @@ files[] = includes/RESTServer.inc
 
 files[] = lib/bencode.php
 files[] = lib/mimeparse.php
-files[] = lib/spyc.php
+files[] = lib/spyc/spyc.php
 
 dependencies[] = services
 
-core = 7.x
\ No newline at end of file
+core = 7.x
