diff --git servers/rest_server/rest_server.module servers/rest_server/rest_server.module
index 6f23fac..5f96237 100755
--- servers/rest_server/rest_server.module
+++ servers/rest_server/rest_server.module
@@ -228,6 +228,6 @@ function _rest_server_get_spyc_location() {
   }
 
   // Libraries not in use fall back to straight look up.
-  $services_spyc_loc = dirname(__FILE__) . '/lib/spyc.php';
+  $services_spyc_loc = dirname(__FILE__) . '/lib/spyc/spyc.php';
   return file_exists($services_spyc_loc) ? $services_spyc_loc : false;
 }
